RecoEgamma/EgammaElectronAlgos/interface/alpaka/helixPropagator.h

```diff
@@ -0,0 +1,230 @@
+
+/**
+ Description: Function to propagate from a point to a surface on the GPU
+*/
+
+#ifndef RecoEgamma_EgammaElectronAlgos_interface_alpaka_helixPropagator_h
+#define RecoEgamma_EgammaElectronAlgos_interface_alpaka_helixPropagator_h
+
+#include <iostream>
+#include <cmath>
+#include <Eigen/Dense>
+
+
+using Vector3D = Eigen::Matrix<float, 3, 1>;
+using Point3D = Eigen::Matrix<float, 3, 1>;
+enum Solution { bothSol, bestSol, onlyPos };
+
+namespace ALPAKA_ACCELERATOR_NAMESPACE {
+
+	namespace Propagators {
+
+		template <typename T>
+		ALPAKA_FN_HOST_ACC ALPAKA_FN_INLINE T sqr(const T& t) {
```

Comment:
`T` is presumably a fundamental type, so
```suggestion
		ALPAKA_FN_HOST_ACC ALPAKA_FN_INLINE T sqr(const T t) {
```
would be more efficient.